src/Symfony/Component/Workflow/WorkflowInterface.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+use Symfony\Component\Workflow\Exception\LogicException;
+use Symfony\Component\Workflow\MarkingStore\MarkingStoreInterface;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+interface WorkflowInterface
+{
+    /**
+     * Returns the object's Marking.
+     *
+     * @param object $subject A subject
+     *
+     * @return Marking The Marking
+     *
+     * @throws LogicException
+     */
+    public function getMarking($subject);
+
+    /**
+     * Returns true if the transition is enabled.
+     *
+     * @param object $subject        A subject
+     * @param string $transitionName A transition
+     *
+     * @return bool true if the transition is enabled
+     *
+     * @throws LogicException
+     */
+    public function can($subject, $transitionName);
+
+    /**
+     * Fire a transition.
+     *
+     * @param object $subject        A subject
+     * @param string $transitionName A transition
+     *
+     * @return Marking The new Marking
+     *
+     * @throws LogicException If the transition is not applicable
+     * @throws LogicException If the transition does not exist
+     */
+    public function apply($subject, $transitionName);
+
+    /**
+     * Returns all enabled transitions.
+     *
+     * @param object $subject A subject
+     *
+     * @return Transition[] All enabled transitions
+     */
+    public function getEnabledTransitions($subject);
+
+    public function getName();
```

Comment:
missing type info +  {@inheritdoc} in sub class (id prefer a `@return string` here).

Reply:
maybe we could typehint it ?